app/models/vm_migrate_workflow.rb

```diff
@@ -36,6 +37,14 @@ def get_source_and_targets(refresh = false)
     else
       add_target(:placement_dc_name, :datacenter, EmsFolder, result)
     end
+
+    unless field_supported(:cluster)
+      # If the user can not pick a cluster there can only be one to select
+      # from => preselect it so the hosts will be filtered accordingly.
+      clusters = vms.map(&:ems_cluster).uniq.compact
```

Comment:
uniq is going to be deprecated use distinct instead.
if you are going to only use the first cluster you can add "first" or "limit(1)" to avoid selecting many you do not need.